pystapi-client/src/pystapi_client/client.py

```diff
@@ -250,9 +259,9 @@ def get_products(self, limit: int | None = None) -> Iterator[ProductsCollection]
         else:
             parameters = {"limit": limit}
 
-        products_collection_iterator = self.stapi_io.get_pages(
-            products_endpoint, parameters=parameters, lookup_key="products"
-        )
+        products_link = Link(href=products_endpoint, method="GET", body=parameters, rel="")
+
+        products_collection_iterator = self.stapi_io.get_pages(link=products_link, lookup_key="products")
```

Comment:
I didn't notice this earlier, but I think `get_products` should return an `Iterator[Product]`. This might be as simple as doing:

```python
for d in products_collection_iterator:
    products_collection = ProductsCollection.model_validate(d)
    yield from products_collection.products
```